dashboard-widgets.el

```diff
@@ -269,6 +271,88 @@ If MESSAGEBUF is not nil then MSG is also written in message buffer."
 	  (dashboard-insert--shortcut "p" "Projects:")))
     (error "Projects list depends on 'projectile-mode` to be activated")))
 
+;;
+;; Org Agenda
+;;
+(defun dashboard-insert-agenda-list (list-display-name list)
+  "Render LIST-DISPLAY-NAME title and agenda items from LIST."
+  (when (car list)
+    (insert list-display-name)
+    (mapc (lambda (el)
+            (insert "\n    ")
+            (let ((filename (nth 4 el))
+                  (lineno (nth 3 el))
+                  (title (nth 0 el)))
+              (widget-create 'push-button
+                             :action `(lambda (&rest ignore)
+                                        (let ((buffer (find-file-other-window ,filename)))
+                                          (with-current-buffer buffer
+                                            (goto-char ,lineno)
+                                            )
+                                          (switch-to-buffer buffer)))
+                             :mouse-face 'highlight
+                             :follow-link "\C-m"
+                             :button-prefix ""
+                             :button-suffix ""
+                             :format "%[%t%]"
+                             (format "%s" title)))
+            )
+          list)))
+
+(defun dashboard--timestamp-to-gregorian-date (timestamp)
```

Comment:
I would rather not having a different style for the function declarations. So instead of `--`, let's just go for a single dash.